juju/juju
# 2.7 - Network Device Detection for New LXD Versions

### Checklist


----

## Description of change

The latest release of LXD (3.22) changed the default format of the profile for NIC devices, though the old format is still valid.

Old example:
```yaml
devices:
  eth0:
    name: eth0
    parent: lxdbr0
    type: nic
    nictype: bridged
```
New example:
```yaml
devices:
  eth0:
    name: eth0
    network: lxdbr0
    type: nic
```

This patch updates the LXD dependency to the hash that corresponds with the 3.22 tag, regenerates mocks and accommodates the new profile schema.

One change brought in with the new dependency is certificate generation, which takes and additional argument for whether to include host names in the certificate. This is always passed as `true` by Juju.

## QA steps

Localhost:
- `juju bootstrap lxd new-profile-test --debug --no-gui` should succeed.

Machine-In-Machine:
- `juju bootstrap aws new-profile-test --debug --no-gui --bootstrap-series focal --force --config image-stream=daily`.
- `juju add-machine --series focal`
- `juju add-machine lxd:0`
- Verify that we are using the 2.33 LXD Snap with: `juju run --machine 0 -- snap info lxd`.

## Documentation changes

None.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1867886
